### ISSUE TITLE
Failing to translate expression extensions

### ISSUE THREAD

**Issue by Schahen** [Kotlin/dukat#404]
Consider following code:
```typescript
interface A {}

declare interface AConstructor {
    new(): A;
    ping();
}

declare var Z: AConstructor;
declare class B extends Z {}
```

It will be translated to: 
```kotlin
external interface A

external interface AConstructor {
    fun ping()
}

external var Z: AConstructor

external open class B : Z
```

Which won't compile, failing with error:
```bash
z.module_tmp.kt:25:25: error: unresolved reference: Z
external open class B : Z
```

This is actually because we don't support extending expressions, as they are described in
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-1-6.html#extending-expressions

Actually, in the link above nothing is said about variables, but de-facto they are supported as well, 

Before fixing https://github.com/Kotlin/dukat/issues/387 we are supposed to do something with this. 

The most obvious thing to do is to just substitute Z with AConstructor wherever we encounter it.

**Comment by Schahen** [comment]
It looks like we expect to have:
```kotlin
external interface A {
   companion object {
       fun ping();
   }
}
external var Z: A;

external open class B : A {
   constructor()
}
```

**Comment by JSMonk** [comment]
Fixed in #460
